### PR TITLE
Update base64 format check

### PR DESCRIPTION
**Pull request by phlipper**
* The main change here is adding `\\r` and `\\n` to the `gsub` in the check method. We were seeing a base64-encoded `SAMLResponse` parameter value come through with these extra escape characters and this solves is the issue.
* Extract the regexp for base64 format to a constant value to avoid additional unnecessary object allocations.
* Rename `is_base64?` to `base64_formatted?` since `rubocop` complains about methods starting with `is_`.

### CODE REVIEW

**Comment by pitbulk** [comment]
:+1:
@Lordnibbler @luisvm what do you think?

File: lib/onelogin/ruby-saml/saml_message.rb [hunk 1]
```diff
@@ -73,11 +75,11 @@ def encode(encoded)
...
       #
-      # The function is not strict and does allow newline. This is because some SAML implementations
-      # uses newline in the base64-encoded data, even if they shouldn't have (RFC4648).
-      def is_base64?(string)
-        string = string.gsub(/\r\n/, "").gsub(/\n/, "")
-        string.match(%r{\A(([A-Za-z0-9+/]{4}))*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)\Z})
+      # The function is not strict and allows newlines. This is because some
+      # SAML implementations use newlines in the base64-encoded data even if
+      # they shouldn't (RFC4648).
+      def base64_formatted?(string)
```
Comment (Lordnibbler) [inline lib/onelogin/ruby-saml/saml_message.rb]: can you use YARD style comments here and explicitly document the return value as well?

**Comment by Lordnibbler** [comment]
one small comment but looks good :P

**Comment by Lordnibbler** [comment]
@pitbulk I had a comment here that was not addressed FWIW

**Comment by pitbulk** [comment]
@Lordnibbler Yes, was already solved: 
https://github.com/onelogin/ruby-saml/commit/0552dc16acd8eba7dc14d361384beb261f4b5912

**Pull request merged**